### DISCUSSION TITLE
大量日志提示 "error": "context deadline exceeded"

### DISCUSSION THREAD

**Discussion by Somalia0932** [IrineSistiana/mosdns#253] [General]
![image](https://user-images.githubusercontent.com/42663232/149046964-3162ce6a-1f26-446f-b656-96839ea7f921.png)

不清楚这是什么原因，每个上游dns经测试都是正常的。

[root@vm106 mosdns]# vi config.yaml 
log:
  level: info
  file: './mosdns.log'

plugin:
  - tag: server          # 服务器插件
    type: server
    args:
      entry:
        - mem_cache           # 启用缓存
        - forward_remote      # 转发至 remote
      server:
        - protocol: udp
          addr: 0.0.0.0:53

  - tag: 'mem_cache'
    type: 'cache'
    args:
      size: 819200
      redis: 'redis://10.0.0.106:6379/0'
      lazy_cache_ttl: 86400
      lazy_cache_reply_ttl: 30
      cache_everything: true

  - tag: ''
    type: 'ttl'
    args:
      minimal_ttl: 300
      maximum_ttl: 3600

  - tag: forward_remote
    type: forward
    args:
      upstream:
        - addr: 61.139.2.69
        - addr: 218.6.200.139
        - addr: 119.29.29.29
        - addr: 1.1.1.1
        - addr: 9.9.9.9
        - addr: 168.95.192.1
        - addr: 218.30.118.6
      fastest_ip: true

**Comment by rampageX** [comment]
`redis: 'redis://10.0.0.106:6379/0'`  这行注释掉

**Reply by IrineSistiana** [reply]
~不是上游的问题 是redis超时了。~

**Comment by IrineSistiana** [comment]
应该不是redis问题。还是上游问题。网不稳。具体看issue。#98

虽然是 cache 报的错。但是源头估计还是因为 forward 超时了但是forward没报。这是个bug以后会改。

**Reply by PussAzuki** [reply]
> 应该不是redis问题。还是上游问题。网不稳。具体看issue。#98
> 
> 虽然是 cache 报的错。但是源头估计还是因为 forward 超时了但是forward没报。这是个bug以后会改。

forward 功能之前就有问题，我在你提及的相关issue里提到的现象应该也是这个bug造成的(?)...
为什么我这么说呢？因为我换成 fast_forward 之后日志里几乎没有这个反馈了🤣
用forward的时候安卓这边经常网卡住（表现为有网但是加载不出来，例如禁用掉httpdns的IP的哔哩哔哩安卓端 表现是真的明显）更改之后基本就正常了（但是CDN炸的时候也会这样加载不出来）

**Reply by IrineSistiana** [reply]
> 我在你提及的相关issue里提到的现象应该也是这个bug造成的(?)...

应该不是。这个bug不影响使用。

开个issue详细说说？我不知道你指哪个issue。

**Reply by PussAzuki** [reply]
应该是你新建 #98 的那段时候，我当时询问如何在fast_forward里使用ipv6 dns，后来发现是我打错字了。

当时顺便发了日志截图，里面刷了一堆这个错误，然后你说是网不稳，但我这只有alidns作为上游才有这个问题。

忘了什么时候开始有“b站app用久了就刷不出图”（手动拦截了httpdns），后面发现手机直连 + dns没有任何额外设置的话又没有这个问题，openwrt那边也没报网络问题，然后就尝试使用fast_forward，发现上述问题复现次数几乎为零了（上游服务器和其他配置完全没动） 于是推测是alidns卡到了forward的bug上(?)

**Reply by IrineSistiana** [reply]
好玄学。鸽了鸽了。

~forward是调的adh的包即使有问题我也不好改。于是索性自己搞了fastforward。反正我自己也只用fastforward~

**Comment by goodboy2gb2** [comment]
![Screenshot_20220120-003617](https://user-images.githubusercontent.com/68678656/150174498-510b2631-a7c6-4cf0-9976-eca5e02ee23e.jpg)
同样遇到这个问题，也是用了forward。我试试fast_forward，看还会不会出现

**Reply by miyunda** [reply]
我从`foward`换成`fast_foward`之后日志里的这个错误就不见了。

**Comment by 0xffffharry** [comment]
同样的问题， fast_forward 换成 forward 可解
